### PR TITLE
Update internal_command_doctor.dart GH steps

### PR DESCRIPTION
**Pull request by 0xMALVEE**

### CODE REVIEW

File: cli/lib/command/internal_command_doctor.dart [hunk 1]
```diff
@@ -135,5 +135,70 @@ class InternalCommandDoctor extends InternalCommand {
         );
       }
     }
+
+    // Check gh CLI installation and authentication
+    {
+      bool ghExists;
```
Comment (TarasMazepa) [inline cli/lib/command/internal_command_doctor.dart:141]:
```suggestion
      bool ghExists = false;
```

File: cli/lib/command/internal_command_doctor.dart [hunk 2]
```diff
@@ -135,5 +135,70 @@ class InternalCommandDoctor extends InternalCommand {
...
+      try {
+        ghExists = context
+            .withSilence(true)
+            .command(['gh', '--version'])
+            .announce('Checking if GitHub CLI is installed.')
+            .runSync()
+            .isSuccess();
+      } catch (e) {
+        ghExists = false;
```
Comment (TarasMazepa) [inline cli/lib/command/internal_command_doctor.dart:150]:
```suggestion
```

File: cli/lib/command/internal_command_doctor.dart (hunk 1 above)
Comment (TarasMazepa) [inline cli/lib/command/internal_command_doctor.dart:141]:
```suggestion
      String? ghVersion = null;
```

File: cli/lib/command/internal_command_doctor.dart [hunk 3]
```diff
@@ -135,5 +135,70 @@ class InternalCommandDoctor extends InternalCommand {
...
+    // Check gh CLI installation and authentication
+    {
+      bool ghExists;
+      try {
+        ghExists = context
+            .withSilence(true)
+            .command(['gh', '--version'])
+            .announce('Checking if GitHub CLI is installed.')
+            .runSync()
+            .isSuccess();
```
Comment (TarasMazepa) [inline cli/lib/command/internal_command_doctor.dart:148]:
```suggestion
            .stdout
            .toString();
```

File: cli/lib/command/internal_command_doctor.dart (hunk 2 above)
Comment (TarasMazepa) [inline cli/lib/command/internal_command_doctor.dart:150]:
```suggestion
```

**Review by TarasMazepa — CHANGES_REQUESTED** [review]
Looking good! Let's capture gh version, also please update tests

File: cli/lib/command/internal_command_doctor.dart [hunk 4]
```diff
@@ -135,5 +135,70 @@ class InternalCommandDoctor extends InternalCommand {
...
+      }
+
+      context.printToConsole(
+        '''[${boolToCheckmark(ghExists)}] gh --version # ${ghExists ? "installed" : "not found"}''',
+      );
+
+      if (!ghExists) {
+        context.printToConsole('''    X Install GitHub CLI using:''');
+        context.printToConsole(
+          '''      gh not installed''',
```
Comment (TarasMazepa) [inline cli/lib/command/internal_command_doctor.dart:160]: Let's give people nice install instructions
```suggestion
          '''      https://github.com/cli/cli#installation''',
```

File: cli/lib/command/internal_command_doctor.dart [hunk 5]
```diff
@@ -135,5 +135,70 @@ class InternalCommandDoctor extends InternalCommand {
...
+            .command(['gh', '--version'])
+            .announce('Checking if GitHub CLI is installed.')
+            .runSync()
+            .isSuccess();
+      } catch (e) {
+        ghExists = false;
+      }
+
+      context.printToConsole(
+        '''[${boolToCheckmark(ghExists)}] gh --version # ${ghExists ? "installed" : "not found"}''',
```
Comment (TarasMazepa) [inline cli/lib/command/internal_command_doctor.dart:154]:
```suggestion
        '''[${boolToCheckmark(ghExists)}] gh --version # $ghVersion''',
```

**Comment by 0xMALVEE** [comment]
Closing this pr cuz commit history is rip. 
This pr now will be done here https://github.com/TarasMazepa/stax/pull/372

**Pull request closed without merging**